### PR TITLE
Feature:(issue_1071) Add check for duplicated flags

### LINKED ISSUE #1071: Add optional check for duplicated string and int type flags.
## Checklist

* [y] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [y] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [y] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

Sometimes users can accidently duplicate string type flags and then the latest value we be taken. It is not clear and bad for usability. 

## Solution description

Add an option called Duplicated true\false to flag options. Example:

```
flagFoo = &cli.StringFlag{
	Name: "foo",
	Duplicated: false,
}
```

A detailed description of what you want to happen.

if Duplicated set to false command should fail with error
"flag foo can not be duplicated"

## Describe alternatives you've considered

For now, as a workaround, we're using StringSliceType for flags we don't want to be duplicated and check that slice have only one element in it.

https://github.com/Venafi/vcert/blob/8e1dc13c0afdd615d7f27b80190ddf3a1d07e848/cmd/vcert/commands.go#L105-L120

### PR DESCRIPTION
**Pull request by dearchap**
## What type of PR is this?

_(REQUIRED)_

- documentation
- feature

## What this PR does / why we need it:

_(REQUIRED)_

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1071 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

make test

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

File: flag_impl.go [hunk 1]
```diff
@@ -57,6 +81,8 @@ type FlagBase[T any, C any, VC ValueCreator[T, C]] struct {
 
 	Config C // Additional/Custom configuration associated with this flag type
 
+	NonDuplicated bool // whether this flag can be duplicated on the command line
```
Comment (meatballhat) [inline flag_impl.go:84]: I feel weird about a bool type name that starts with a negation, but I understand we want the default behavior to be to allow multiple instances. What do you think about:

- `StrictOnce`
- `StrictlyOnce`
- `Once`
- `OnlyOnce`

?

Comment (abitrolly) [reply flag_impl.go:84]: * `Repeatable`

?

Comment (dearchap) [reply flag_impl.go:84]: I think OnlyOnce is good since by default it would be false.

**Review by meatballhat — APPROVED** [review]
Yay! :tada:

**Pull request merged**